httpload/cmd/httpload/main.go

```diff
@@ -0,0 +1,319 @@
+	lt.dumpTimings(w)
+
+	fmt.Fprintf(w, "\n%s\n\n", "Status code distribution:")
+	lt.dumpStatuses(w)
+}
+
+func (lt *LoadTester) run(config Config) error {
+	var (
+		ctx    context.Context
+		cancel context.CancelFunc
+	)
+
+	if config.appDuration > 0 {
+		ctx, cancel = context.WithTimeout(context.Background(), config.appDuration)
+	} else {
+		ctx, cancel = context.WithCancel(context.Background())
+	}
+	defer cancel()
+
+	return lt.makeLoadRequest(ctx, config.url)
+}
+
+func (lt *LoadTester) makeLoadRequest(ctx context.Context, query string) error {
+	req, err := http.NewRequest("GET", query, nil)
+	if err != nil {
+		return err
+	}
+
+	start := time.Now()
+	err = lt.httpDo(ctx, req, func(resp *http.Response, err error) error {
+		if err != nil {
+			return err
+		}
+
+		defer resp.Body.Close()
+
+		elapsed := time.Since(start)
+		code := resp.StatusCode
+
+		lt.results = append(lt.results, Result{code, elapsed})
+
+		return nil
+	})
+
+	return err
+}
+
+func (lt *LoadTester) httpDo(ctx context.Context, req *http.Request, f func(*http.Response, error) error) error {
+	c := make(chan error, 1)
+	req = req.WithContext(ctx)
+	go func() { c <- f(http.DefaultClient.Do(req)) }()
+	select {
+	case <-ctx.Done():
+		<-c // Wait for f to return.
+		return ctx.Err()
+	case err := <-c:
+		return err
+	}
+}
+
+func (lt *LoadTester) dumpStatuses(w io.Writer) {
+	statuses := make(map[int]int)
+
+	for _, result := range lt.results {
+		statuses[result.status]++
+	}
+
+	keys := make([]int, 0, len(statuses))
+
+	for key := range statuses {
+		keys = append(keys, key)
+	}
+
+	sort.Ints(keys)
+
+	for _, key := range keys {
+		code := float64(key)
+		count := float64(statuses[key])
+		fmt.Fprintf(w, "%s\n", fmt.Sprintf("[%3.f] %12.f response(s)", code, count))
+	}
+}
+
+func (lt *LoadTester) dumpTimings(w io.Writer) {
+	totalTiming := float64(0)
+	maxTiming := float64(0)
+	minTiming := float64(0)
+	averageTiming := float64(0)
+	requestsPerSecond := float64(0)
+
+	n := len(lt.results)
+	noResults := (n == 0)
+
+	if noResults {
+		return
+	}
+
+	timing0 := lt.results[0].timing.Seconds()
+	totalTiming = timing0
+	maxTiming = timing0
+	minTiming = timing0
+
+	for i := 1; i < n; i++ {
+		timing := lt.results[i].timing.Seconds()
+
+		totalTiming += timing
+
+		if timing > maxTiming {
+			maxTiming = timing
+		}
+
+		if timing < minTiming {
+			minTiming = timing
+		}
+	}
+
+	averageTiming = totalTiming / float64(n)
+	requestsPerSecond = float64(1) / averageTiming
+
+	fmt.Fprintf(w, "%s\n", fmt.Sprintf("Total:        %12.4f secs", totalTiming))
+	fmt.Fprintf(w, "%s\n", fmt.Sprintf("Slowest:      %12.4f secs", maxTiming))
+	fmt.Fprintf(w, "%s\n", fmt.Sprintf("Fastest:      %12.4f secs", minTiming))
+	fmt.Fprintf(w, "%s\n", fmt.Sprintf("Average:      %12.4f secs", averageTiming))
+	fmt.Fprintf(w, "%s\n", fmt.Sprintf("Requests/sec: %12.4f", requestsPerSecond))
+}
+
+func main() {
+	flags := flag.NewFlagSet(os.Args[0], flag.ContinueOnError)
```

Comment:
At the moment, calling `httpload -h` has not the expected effect.
Just changing the constant to `flag.ExitOnError` would solve that